tests/http-adapter.spec.js

```diff
@@ -33,166 +29,252 @@ describe('HTTP adapter', function () {
+        let calls = fetchMocked.calls()
+        expect(calls).toHaveLength(1)
+        expect(calls[0][0].includes(path1)).toBe(true)
+        expect(adapter._outstandingRequests.size).toEqual(1)
+      })
+      adapter.get(path2).then(() => {
+        let calls = fetchMocked.calls()
+        expect(calls).toHaveLength(2)
+        expect(calls[1][0].includes(path2)).toBe(true)
+        expect(adapter._outstandingRequests.size).toEqual(0)
+        done()
+      })
     })
 
-    it('parallel requests are restored after `serializeRequests` is toggled back', async function () {
-      server.autoRespond = false
-      server.respondImmediately = false
+    it('parallel requests do not start when `serializeRequests = true`, even if it was set after the pending request started', async done => {
+      const fetchMocked = fetchMock.sandbox().mock('*', {
+        status: 200,
+        method: 'GET',
+        body: {}
+      })
+      let numCalled = 0
+      
+      global.fetch = async function (url, init) {
+        const callOrder = numCalled++
+
+        switch (callOrder) {
+          case 0:
+            expect(url.pathname.includes(path1)).toBe(true)
+            //current outstanding requests should be zero since serializeRequests was turned on AFTER the request started
+            expect(adapter._outstandingRequests.size).toEqual(0)
+            break
+          case 1:
+            expect(url.pathname.includes(path2)).toBe(true)
+            expect(adapter._outstandingRequests.size).toEqual(1)
+            break
+        }
+
+        return await fetchMocked(url, init)
+      }
 
-      // Turn on `serializeRequests` and make some requests
+      // Turn on `serializeRequests` and then make two parallel requests
+      
+      adapter.get(path1).then(() => {
+        let calls = fetchMocked.calls()
+        expect(calls).toHaveLength(1)
+        expect(calls[0][0].includes(path1)).toBe(true)
+        expect(adapter._outstandingRequests.size).toEqual(1)
+      })
       adapter.serializeRequests = true
-      const fetch1 = adapter.get('/api/user/42/')
-      const fetch2 = adapter.get('/api/offer/9000/')
+      adapter.get(path2).then(() => {
+        let calls = fetchMocked.calls()
+        expect(calls).toHaveLength(2)
+        expect(calls[1][0].includes(path2)).toBe(true)
+        expect(adapter._outstandingRequests.size).toEqual(0)
+        done()
+      })
+    })
 
-      // Respond to requests
-      await Promise.resolve().then(() => {})
-      const payload = {foo: 'bar', fiz: {biz: 'buz'}}
-      server.requests[0].respond(200, { "Content-Type": "application/json" }, JSON.stringify(payload))
-      await fetch1
+    it('parallel requests are restored after `serializeRequests` is toggled back', async done => {
+      const fetchMocked = fetchMock.sandbox().mock('*', {
+        status: 200,
+        method: 'GET',
+        body: {}
+      })
+      let numCalled = 0
+      
+      global.fetch = async function (url, init) {
```

Comment:
Like we talked about f2f, let's replace this with `jest.fn` et al. No need to re-implement mocking and spying.

Reply:
To get you started, what you've written so far is roughly equivalent to:
```js
global.fetch = jest.fn()
  .mockResolvedValue({}) // default
  .mockResolvedValueOnce(firstResponse)
  .mockResolvedValueOnce(secondResponse)
```
etc.

But the `fetchMock` library probably has a little more depth to explore here, too.